sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CacheTableExec.scala

```diff
@@ -94,14 +94,19 @@ case class CacheTableAsSelectExec(
   override lazy val relationName: String = tempViewName
 
   override lazy val planToCache: LogicalPlan = {
+    // If the plan cannot be analyzed, throw an exception and don't proceed.
+    val qe = sparkSession.sessionState.executePlan(query)
+    qe.assertAnalyzed()
+    val analyzedPlan = qe.analyzed
```

Comment:
@cloud-fan If we make `query` as `child` of `CacheTableAsSelectExec`, it will be an optimized plan instead of an analyzed plan, which can be problematic when looking up the cached plan. One way to work around is to move matching `CacheTableAsSelect` back to `ResolveSessionCatalog`... Any thought?

Reply:
The current code looks fine.  I think `CacheTableAsSelectExec` is the only exception that it has a `query` which is not a simple table relation but we want to skip optimizing it. Let's document this clearly.

Reply:
thanks, updated.